services/app-api/src/dataMigrations/migrations/20231106200334_fix_empty_rates.ts

```diff
@@ -0,0 +1,85 @@
+import type { PrismaTransactionType } from '../../postgres/prismaTypes'
+
+export const migrate = async (
+    tx: PrismaTransactionType
+): Promise<Error | undefined> => {
+    try {
+        const initialRates = await tx.rateTable.findMany({
+            where: {
+                revisions: {
+                    some: {
+                        submitInfo: {
+                            isNot: null,
+                        },
+                    },
+                },
+                stateCode: {
+                    not: 'AS', // exclude test state as per ADR 019
+                },
+            },
+        })
+        console.info(
+            ` ---- Prepare to migrate. Currently there are ${initialRates.length} rates`
+        )
+
+        const badRateRevisions = await tx.rateRevisionTable.findMany({
+            where: {
+                rateDateStart: null,
+                rateDateEnd: null,
+                rateDateCertified: null,
```

Comment:
I will try adding that back. I had that constraint in there and it wasn't deleting the ones I wanted but since then I simplified things more. Let me try adding the contract revisions are contract only thing back

Reply:
I've added these adjustments back to the query FYI